### PR TITLE
Docs: Add warning against installing Ray in base environment

### PR DESCRIPTION
**Pull request by MysteryManav**
Docs: Added warning in documentation against installing Ray in base environment

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Comment by MysteryManav** [comment]
@Michaelvll if you would, please take a look at it and notify for any additional changes

File: docs/source/running-jobs/distributed-jobs.rst [hunk 1]
```diff
@@ -137,3 +137,14 @@ This allows you directly to SSH into the worker nodes, if required.
   # Worker nodes.
   $ ssh mycluster-worker1
   $ ssh mycluster-worker2
+
+
+Executing a Distributed Ray Program
```
Comment (concretevitamin) [inline docs/source/running-jobs/distributed-jobs.rst:142]: For discussion: We may want to add a dedicated guide on how to use Ray as part of the user job. That page can use a tip like this. A full YAML example https://github.com/skypilot-org/skypilot/issues/3195 may help promote this recommendation more directly.

Comment (Michaelvll) [reply docs/source/running-jobs/distributed-jobs.rst:142]: Yes, let’s add a yaml example of starting a distributed job with ray as mentioned above. We can try to move it to a separate page later after this pr is merged.

Comment (MysteryManav) [reply docs/source/running-jobs/distributed-jobs.rst:142]: So for confirmation: I should add the command to run the example in distributed_ray_train, the one that uses FashionMNIST and add the command for tha yaml file and its output as example in the documentation ?

Comment (Michaelvll) [reply docs/source/running-jobs/distributed-jobs.rst:142]: Yes, we can add the yaml file inline with the command to run that [yaml](https://github.com/skypilot-org/skypilot/blob/master/examples/distributed_ray_train/ray_train.yaml) : )

**Comment by MysteryManav** [comment]
any more suggestions ?

File: docs/source/running-jobs/distributed-jobs.rst [hunk 2]
```diff
@@ -137,3 +137,48 @@ This allows you directly to SSH into the worker nodes, if required.
...
+------------------------------------
+To execute a distributed Ray program on many VMs, you can use the following example:
+
+.. code-block:: console
+
+  $ sky launch ray_train.yaml
+
+.. code-block:: yaml
+    :emphasize-lines: 6-6,21-22,24-25
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:151]: It seems the emphasize lines seems unrelated here. Should we just remove it?
```suggestion
```

File: docs/source/running-jobs/distributed-jobs.rst [hunk 3]
```diff
@@ -137,3 +137,48 @@ This allows you directly to SSH into the worker nodes, if required.
...
+  
+    resources:
+      accelerators: L4:2
+      memory: 64+
+  
+    num_nodes: 2
+  
+    setup: |
+      pip install "ray[train]"
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:160]: Let's keep the original conda commands to follow the warning below.
```suggestion
      conda activate ray
      if [ $? -ne 0 ]; then
        conda create -n ray python=3.10 -y
        conda activate ray
      fi
      
      pip install "ray[train]"
```

File: docs/source/running-jobs/distributed-jobs.rst [hunk 4]
```diff
@@ -137,3 +137,48 @@ This allows you directly to SSH into the worker nodes, if required.
   # Worker nodes.
   $ ssh mycluster-worker1
   $ ssh mycluster-worker2
+
+
+Executing a Distributed Ray Program
+------------------------------------
+To execute a distributed Ray program on many VMs, you can use the following example:
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:144]:
```suggestion
To execute a distributed Ray program on many VMs, you can download the `training script <https://github.com/skypilot-org/skypilot/blob/master/examples/distributed_ray_train/train.py>`_ and launch the `task yaml <https://github.com/skypilot-org/skypilot/blob/master/examples/distributed_ray_train/ray_train.yaml>`_:
```

File: docs/source/running-jobs/distributed-jobs.rst [hunk 5]
```diff
@@ -137,3 +137,48 @@ This allows you directly to SSH into the worker nodes, if required.
...
+
+.. code-block:: console
+
+  $ sky launch ray_train.yaml
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:148]:
```suggestion

  $ wget https://raw.githubusercontent.com/skypilot-org/skypilot/master/examples/distributed_ray_train/train.py
  $ sky launch ray_train.yaml
```

File: docs/source/running-jobs/distributed-jobs.rst [hunk 6]
```diff
@@ -137,3 +137,48 @@ This allows you directly to SSH into the worker nodes, if required.
...
+.. code-block:: yaml
+    :emphasize-lines: 6-6,21-22,24-25
+  
+    resources:
+      accelerators: L4:2
+      memory: 64+
+  
+    num_nodes: 2
+  
+    setup: |
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:159]:
```suggestion

    workdir: .

    setup: |
```

File: docs/source/running-jobs/distributed-jobs.rst [hunk 7]
```diff
@@ -137,3 +137,48 @@ This allows you directly to SSH into the worker nodes, if required.
...
+        sleep 5
+        ps aux | grep ray | grep 6379 &> /dev/null || ray start --address $head_ip:6379 --disable-usage-stats
+      fi
+
+.. warning:: 
+  **Avoid Installing Ray in Base Environment**
+
+  Before proceeding with the execution of a distributed Ray program, it is crucial to ensure that Ray is **not** installed in the base environment. Installing a different version of Ray in the base environment can lead to compatibility issues, conflicts, and unintended consequences.
+
+  To maintain a clean and stable environment for your distributed Ray program, it is highly recommended to **create a dedicated virtual environment** for Ray and its dependencies. This helps isolate the Ray installation and prevents interference with other packages in your base environment.
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:183]:
```suggestion
  **Avoid Installing Ray in Base Environment**: Before proceeding with the execution of a distributed Ray program, it is crucial to ensure that Ray is **not** installed in the *base* environment. Installing a different version of Ray in the base environment can lead to abnormal cluster status.

  It is highly recommended to **create a dedicated virtual environment** (as above) for Ray and its dependencies, and avoid calling `ray stop` as that will also cause issue with the cluster.
```

**Review by Michaelvll — COMMENTED** [review]
Thanks for fixing this @MysteryManav! Just left several comments.

**Comment by MysteryManav** [comment]
thankful for the opportunity and the guidance @Michaelvll

File: docs/source/running-jobs/distributed-jobs.rst [hunk 8]
```diff
@@ -137,3 +137,60 @@ This allows you directly to SSH into the worker nodes, if required.
...
+      
+      pip install "ray[train]"
+      pip install tqdm
+      pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
+  
+    run: |
+      sudo chmod 777 -R /var/tmp
+      head_ip=`echo "$SKYPILOT_NODE_IPS" | head -n1`
+      num_nodes=`echo "$SKYPILOT_NODE_IPS" | wc -l`
+      ray start --head  --disable-usage-stats --port 6379
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:182]: This line shouldbe removed : )
```suggestion
```

**Review by Michaelvll — APPROVED** [review]
Thanks for updating the doc @MysteryManav! This seems good to me as a basic version. We should probably have a more detailed explanation for those concept, e.g. the relationship with ray, etc.
cc'ing @concretevitamin for another look.

File: docs/source/running-jobs/distributed-jobs.rst [hunk 9]
```diff
@@ -137,3 +137,59 @@ This allows you directly to SSH into the worker nodes, if required.
...
+
+    workdir: .
+
+    setup: |
+      conda activate ray
+      if [ $? -ne 0 ]; then
+        conda create -n ray python=3.10 -y
+        conda activate ray
+      fi
+
```
Comment (Michaelvll) [inline docs/source/running-jobs/distributed-jobs.rst:167]:
```suggestion
```

Duplicated?

**Review by concretevitamin — APPROVED** [review]
Awesome, LGTM too @MysteryManav!

**Comment by MysteryManav** [comment]
@concretevitamin @Michaelvll  just commited a last suggestion, so i think it needs to be approved again
